protocol/statetracker/updaters/policy_updater.go

```diff
@@ -0,0 +1,92 @@
+package updaters
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/lavanet/lava/protocol/lavasession"
+	"github.com/lavanet/lava/utils"
+	plantypes "github.com/lavanet/lava/x/plans/types"
+)
+
+const (
+	CallbackKeyForPolicyUpdate = "policy-update"
+)
+
+type PolicySetter interface {
+	SetPolicy(policyInformation map[string]struct{}) error
+	BuildMapFromPolicyQuery(policy *plantypes.Policy, chainId string, apiInterface string) (map[string]struct{}, error)
+}
+
+type PolicyFetcher interface {
+	GetConsumerPolicy(ctx context.Context, consumerAddress, chainID string) (*plantypes.Policy, error)
+}
+
+type PolicyUpdater struct {
+	lock                  sync.RWMutex
+	chainId               string
+	consumerAddress       string
+	lastTimeUpdatedPolicy uint64
+	policyFetcher         PolicyFetcher
+	policyUpdatables      map[string]PolicySetter // key is apiInterface.
+}
+
+func NewPolicyUpdater(chainId string, policyFetcher PolicyFetcher, consumerAddress string, policyUpdatable PolicySetter, endpoint lavasession.RPCEndpoint) *PolicyUpdater {
+	return &PolicyUpdater{
+		chainId:               chainId,
+		policyFetcher:         policyFetcher,
+		policyUpdatables:      map[string]PolicySetter{endpoint.ApiInterface: policyUpdatable},
+		consumerAddress:       consumerAddress,
+		lastTimeUpdatedPolicy: 0,
+	}
+}
+
+func (pu *PolicyUpdater) AddPolicySetter(policyUpdatable PolicySetter, endpoint lavasession.RPCEndpoint) error {
+	pu.lock.Lock()
+	defer pu.lock.Unlock()
+	existingPolicySetter, found := pu.policyUpdatables[endpoint.ApiInterface]
+	if found {
+		return utils.LavaFormatError("Trying to register to policy updates on already registered api interface", nil,
+			utils.Attribute{Key: "endpoint", Value: endpoint},
+			utils.Attribute{Key: "policyUpdatable", Value: existingPolicySetter})
+	}
+	pu.policyUpdatables[endpoint.ApiInterface] = policyUpdatable
+	return nil
+}
+
+func (pu *PolicyUpdater) UpdaterKey() string {
+	return CallbackKeyForPolicyUpdate + pu.chainId
+}
+
+func (pu *PolicyUpdater) BuildPolicyMapAndSetPolicy(policyUpdatable PolicySetter, policy *plantypes.Policy, apiInterface string) error {
+	policyMap, err := policyUpdatable.BuildMapFromPolicyQuery(policy, pu.chainId, apiInterface)
```

Comment:
why do we need two functions here, we can just do SetPolicy in the policyUpdatable, it's not like we are using the policy map